velox/dwio/parquet/writer/CMakeLists.txt

```diff
@@ -12,7 +12,16 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+add_subdirectory(arrow)
+
 add_library(velox_dwio_arrow_parquet_writer Writer.cpp)
 
-target_link_libraries(velox_dwio_arrow_parquet_writer velox_dwio_common
-                      velox_arrow_bridge parquet arrow fmt::fmt)
+target_link_libraries(
+  velox_dwio_arrow_parquet_writer
+  velox_dwio_arrow_parquet_writer_lib
+  velox_dwio_arrow_parquet_writer_util_lib
+  velox_dwio_common
+  velox_arrow_bridge
+  parquet
```

Comment:
Can we remove the `parquet` dependency coming from the Arrow library?
It will be cleaner if we can achieve this.
We can then disable PARQUET via `ARROW_PARQUET=OFF` in Arrow build and remove the Thrift dependency.

Reply:
We will just need `libarrow` for the core.

Reply:
Let me double check, but I think we need to move a couple more things before we can remove it completely. Since I got to a stage where things compile file and tests pass, my thinking is that this is a good checkpoint so we don't make this PR even larger. But I'll keep moving stuff until we can remove the dependency.

Reply:
Sounds good to me.